dds/idl/anonymous.h

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef anonymous_H
+#define anonymous_H
+
+#include "dds_generator.h"
+#include "utl_scoped_name.h"
+#include "ast.h"
+
+#include <string>
+
+struct Field {
```

Comment:
This header only defines a class called `Field`, but the header is not `field.h`.  I think we should find a different name for `Field`.  Is it used for all Fields or just sequence/array fields?

Reply:
It was AnonymousField for anonymous sequence/array. Now it handles non-anonymous fields so the generation code (marshal, meta class, etc) can be shared. And hopefully, the c++11 mapping can be done in one place here for all.

Reply:
My comment about the file name vs. the class name stands.